daemon/snap.go

```diff
@@ -339,12 +340,21 @@ func fillComponentInfo(about aboutSnap) []client.Component {
 	return comps
 }
 
-// snapIcon tries to find the icon inside the snap
-func snapIcon(info snap.PlaceInfo) string {
+// snapIcon tries to find the icon inside the snap at meta/gui/icon.*, and if
+// the snap does not ship an icon there, then tries to find the fallback icon
+// in the icons install directory.
+func snapIcon(info snap.PlaceInfo, snapID string) string {
+	// Look in the snap itself
 	found, _ := filepath.Glob(filepath.Join(info.MountDir(), "meta", "gui", "icon.*"))
-	if len(found) == 0 {
-		return ""
+	if len(found) > 0 {
+		return found[0]
```

Comment:
Maybe it would be better to prioritize .svg over .png, in the case that both do exist?